src/main/java/com/w6/nlp/Parser.java

```diff
@@ -117,10 +145,84 @@ public Response generateResponse(final String input) {
             when.addAll(sentenseWhen);
             weapon.addAll(sentenseWeapon);
         }
+        
+        
 
+        removeEquals(who);
+        removeEquals(what);
+        removeEquals(whom);
+        removeEquals(weapon);
+        
+        
+        removeAndCountRatedEquals(ratedWhen,when);
+        removeAndCountRatedEquals(ratedWhere,where);
 
 
         
         return new Response(text, new Table(who, weapon, what, whom, where, when));
     }
+    
+    private void removeEquals(
+            List<String> list
+    ) {
+        
+       Set<String> all = new HashSet<>(list);
+       list.clear();
+       list.addAll(all);
+        
+    }
+    
+    private void removeAndCountRatedEquals(List<Pair<String,Integer>> ratedList, List<String> list)
+    {
+        Map<String, Pair<Integer,Integer>> statMap = new HashMap<>();
+        for(Pair<String,Integer> pair : ratedList)
+        {
+            String word = pair.first;
+            Integer value = pair.second;
+
+            if(statMap.containsKey(word)){
+                statMap.replace(word,
+                        new Pair<>(
+                                statMap.get(word).first+1,
+                                statMap.get(word).second + value
+                        )
+                );
+            } else {
+                statMap.put(
+                        word,
+                        new Pair<>(1, value)
+                );
+            }
+        }
+        ratedList.clear();
+
+        ArrayList<Pair<String, Double>> values = new ArrayList<>();
+
+        for(String string : statMap.keySet()){
+            Pair<Integer, Integer> val = statMap.get(string);
+            values.add(new Pair<>(string, (1.0 *val.second)/val.first));
```

Comment:
You normalize it on number of time you find it in a text. I think if word appears multiple times it's better than word that apper one time.